### PR TITLE
Homework1

### PR DESCRIPTION
**Pull request by artemd24**
Solutions for tasks from the first homework

### CODE REVIEW

**Comment by apaniukov** [comment]
Hi! Could you add an action that checks formatting, do linting and runs tests for every PR (or every push)?

File: homework1/task02.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+We guarantee, that the given sequence contain >= 0 integers inside.
+
+"""
+from typing import Sequence
+
+
+def check_fibonacci(data: Sequence[int]) -> bool:
+    if (
+        len(data) == 0
```
Comment (apaniukov) [inline homework1/task02.py]: I would argue that if the sequence contains no elements it is a Fibonacci sequence. You just can't disprove it, can you?

**Comment by artemd24** [comment]
> Hi! Could you add an action that checks formatting, do linting and runs tests for every PR (or every push)?

Hello! Sure, I will add it

File: homework1/tests/test_task3.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+from homework1.task03 import find_maximum_and_minimum
+
+
+@pytest.mark.parametrize(
+    ["file_name", "expected_result"],
+    [
+        ("tests/test_data/task3_test1.txt", (-1000, 1000)),
+        ("tests/test_data/task3_test2.txt", (-15, 14)),
+        ("tests/test_data/task3_test3.txt", (0, 0)),
```
Comment (apaniukov) [inline homework1/tests/test_task3.py]: What could be done to not test the function with predefined files (they could be large and not suitable to push to the repository or you might came up with huge number of test cases)?

File: homework1/task04.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+def check_sum_of_four(a: List[int], b: List[int], c: List[int], d: List[int]) -> int:
+    counter = 0
+
+    for i in a:
+        for j in b:
+            for k in c:
+                for l in d:
+                    if i + j + k + l == 0:
+                        counter += 1
```
Comment (apaniukov) [inline homework1/task04.py]: What if you cannot use `if` here?

File: homework1/tests/test_task3.py (hunk 2 above)
Comment (artemd24) [reply homework1/tests/test_task3.py]: 🤔

File: homework1/task02.py (hunk 1 above)
Comment (artemd24) [reply homework1/task02.py]: Yes, I guess that empty sequence is a subsequence for every sequence, I will fix it

File: homework1/task04.py (hunk 3 above)
Comment (artemd24) [reply homework1/task04.py]: Hmm, yes, I can write something like this
```Python
counter += not bool(i + j + k + l)
```
I will rewrite it

File: homework1/tests/test_task3.py (hunk 2 above)
Comment (artemd24) [reply homework1/tests/test_task3.py]: I added random numbers generator for the file

File: homework1/task02.py (hunk 1 above)
Comment (artemd24) [reply homework1/task02.py]: Done

File: homework1/task04.py (hunk 3 above)
Comment (artemd24) [reply homework1/task04.py]: Done

**Comment by artemd24** [comment]
> > Hi! Could you add an action that checks formatting, do linting and runs tests for every PR (or every push)?
> 
> Hello! Sure, I will add it

Done

File: homework1/tests/test_task3.py (hunk 2 above)
Comment (artemd24) [reply homework1/tests/test_task3.py]: Done

Comment (apaniukov) [reply homework1/tests/test_task3.py]: It is not exactly what I meant. You can split the task into two functions - one reads a file and the other finds min and max. This way you can test the main functionality without creating any files.

File: homework1/task04.py (hunk 3 above)
Comment (apaniukov) [reply homework1/task04.py]: That is one way to do it, but not that readable. Something like `counter += (i + j + k + l == 0)` is better. And the best way to solve this problem is to use `itertools`.

**Review by apaniukov — APPROVED** [review]

File: homework1/task04.py (hunk 3 above)
Comment (artemd24) [reply homework1/task04.py]: Make with using itertools

**Pull request merged**